src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -33,14 +33,16 @@ internal class AzureAppConfigurationProvider : ConfigurationProvider, IConfigura
         private ConcurrentDictionary<KeyValueIdentifier, ConfigurationSetting> _watchedSettings = new ConcurrentDictionary<KeyValueIdentifier, ConfigurationSetting>();
 
         private readonly TimeSpan MinCacheExpirationInterval;
-        private readonly SemaphoreSlim InitializationSemaphore = new SemaphoreSlim(1);
 
         // The most-recent time when the refresh operation attempted to load the initial configuration
         private DateTimeOffset InitializationCacheExpires = default;
 
         private static readonly TimeSpan MinDelayForUnhandledFailure = TimeSpan.FromSeconds(5);
         private static readonly TimeSpan DefaultMaxSetDirtyDelay = TimeSpan.FromSeconds(30);
 
+        // To avoid concurrent network operations, this flag is used to achieve synchronization between multiple threads.
+        private static int _networkOperationsInProgress = 0;

```

Comment:
>static [](http://example.com/codeflow?start=16&length=6)

Why is it static? Two different providers should be able to operate independently.

Reply:
You're right. It shouldn't be static.